packages/app/src/app/pages/Dashboard/types.ts

```diff
@@ -85,6 +86,25 @@ export type DashboardNewRepo = {
   type: 'new-repo';
 };
 
+export type DashboardBetaNewRepo = {
+  type: 'beta-new-repo';
+};
+
+type Unpacked<T> = T extends (infer U)[]
+  ? U
+  : T extends (...args: any[]) => infer U
+  ? U
+  : T extends Promise<infer U>
+  ? U
+  : T;
```

Comment:
What does this do? Looks magical 😄

Reply:
It extracts the type of the items from an array, for example
```tsx
Unpacked<string[]> // string
Unpacked<(number | string )[]> // number | string 
```